pkg/addon/addon.go

```diff
@@ -1098,50 +1111,52 @@ func (h *Installer) dispatchAddonResource(addon *InstallPackage) error {
 	if err := passDefInAppAnnotation(defs, app); err != nil {
 		return errors.Wrapf(err, "cannot pass definition to addon app's annotation")
 	}
-
-	if err = h.createOrUpdate(app); err != nil {
-		return err
-	}
-
-	for _, def := range defs {
-		if !checkBondComponentExist(*def, *app) {
-			continue
-		}
-		// if binding component exist, apply the definition
-		addOwner(def, app)
-		err = h.apply.Apply(h.ctx, def, apply.DisableUpdateAnnotation())
+	if h.dryRun {
+		result, err := yaml.Marshal(app)
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "dry-run marshal app into yaml %s", app.Name)
 		}
-	}
-
-	for _, schema := range schemas {
-		addOwner(schema, app)
-		err = h.apply.Apply(h.ctx, schema, apply.DisableUpdateAnnotation())
+		h.dryRunBuff.Write(result)
+		h.dryRunBuff.WriteString("\n")
+	} else {
+		err = h.createOrUpdate(app)
 		if err != nil {
 			return err
 		}
 	}
 
-	for _, view := range views {
-		addOwner(view, app)
-		err = h.apply.Apply(h.ctx, view, apply.DisableUpdateAnnotation())
-		if err != nil {
-			return err
-		}
-	}
+	auxiliaryOutputs = append(auxiliaryOutputs, defs...)
+	auxiliaryOutputs = append(auxiliaryOutputs, schemas...)
+	auxiliaryOutputs = append(auxiliaryOutputs, views...)
 
 	for _, o := range auxiliaryOutputs {
+		// bind-component means the content is related with the component
+		// if component not exists, the resources shouldn't be applied
 		if !checkBondComponentExist(*o, *app) {
 			continue
 		}
+		if h.dryRun {
+			result, err := yaml.Marshal(o)
+			if err != nil {
+				return errors.Wrapf(err, "dry-run marshal auxiliary object into yaml %s", o.GetName())
+			}
+			h.dryRunBuff.WriteString("---\n")
+			h.dryRunBuff.Write(result)
+			h.dryRunBuff.WriteString("\n")
+			continue
+		}
 		addOwner(o, app)
 		err = h.apply.Apply(h.ctx, o, apply.DisableUpdateAnnotation())
 		if err != nil {
 			return err
 		}
 	}
 
+	if h.dryRun {
+		fmt.Print(h.dryRunBuff.String())
```

Comment:
It is more well to support defining the output way.

Reply:
what do you mean by defining the output way?

Reply:
`yaml` or `json`?

Reply:
Users can use like this:

```
vela addon enable velaux --dry-run > test.yaml
```

or 

```
vela addon enable velaux --dry-run | kubectl apply -f -
```

Reply:
Will other logs become noise?

Reply:
no, I've tested and it works.

![image](https://user-images.githubusercontent.com/2173670/190976037-f17e22f0-6208-482a-a5ae-ce2364898348.png)
![image](https://user-images.githubusercontent.com/2173670/190976075-b63ed021-52de-44a9-aebb-36e2fc19bae7.png)